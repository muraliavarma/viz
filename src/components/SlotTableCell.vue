<script>
export default {
  name: 'SlotTableCell',
  functional: true,
  props: ['row', 'rowIndex', 'columnIndex', 'data'],
  render(createElement, ctx) {
    const {
      row, rowIndex, columnIndex, data,
    } = ctx.data.props;
    const { cell } = data.props;
    let renderedCell;
    if (typeof cell === 'function') {
      renderedCell = cell({
        row,
        rowIndex,
        columnIndex,
      });
    } else {
      renderedCell = cell;
    }

    // each cell needs its own unique key since these could be populated dynamically
    return createElement(
      'td',
      Object.assign({}, {
        key: Math.random().toString(36).substr(2),
      }, data), [renderedCell],
    );
  },
};
</script>
