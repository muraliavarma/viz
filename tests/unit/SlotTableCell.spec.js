import { mount } from '@vue/test-utils';
import SlotTableCell from '@/components/SlotTableCell.vue';

describe('SlotTableCell.vue', () => {
  it('outputs the slot', () => {
    const cell = 'abc';
    const wrapper = mount(SlotTableCell, {
      context: {
        props: {
          row: 'a',
          rowIndex: 'b',
          columnIndex: 'c',
          data: { props: { cell } },
        },
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('calls the slot scoped function', () => {
    const cell = ({ row, rowIndex, columnIndex }) => `${row}-${rowIndex}-${columnIndex}`;
    const wrapper = mount(SlotTableCell, {
      context: {
        props: {
          row: 'a',
          rowIndex: 'b',
          columnIndex: 'c',
          data: { props: { cell } },
        },
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
