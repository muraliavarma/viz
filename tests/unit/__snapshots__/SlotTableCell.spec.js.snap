// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlotTableCell.vue calls the slot scoped function 1`] = `
<td>
  a-b-c
</td>
`;

exports[`SlotTableCell.vue outputs the slot 1`] = `
<td>
  abc
</td>
`;
